p2p/src/peers/mod.rs

```diff
@@ -213,39 +214,50 @@ impl Peers {
     }
 
     /// Get a random socket address from the peers list
-    pub fn get_random(&self) -> Result<Option<SocketAddr>, failure::Error> {
-        let bucket = match (self.new_bucket.is_empty(), self.tried_bucket.is_empty()) {
-            (true, true) => return Ok(None),
-            (true, false) => &self.tried_bucket,
-            (false, true) => &self.new_bucket,
-            (false, false) => {
-                if thread_rng().gen() {
-                    &self.tried_bucket
-                } else {
-                    &self.new_bucket
-                }
-            }
-        };
+    pub fn get_random_peers(&self, n: usize) -> Result<Vec<SocketAddr>, failure::Error> {
```

Comment:
This function doesn't need to be this complex. One-liner version using iterators:

```suggestion
    pub fn get_random_peers(&self, n: usize) -> Result<Vec<SocketAddr>, failure::Error> {
        Ok(self
            .new_bucket
            .values()
            .chain(self.tried_bucket.values())
            .map(|p| p.address)
            .choose_multiple(&mut rand::thread_rng(), n))
    }
```

Reply:
This kind of solution not provides the same probability to the different buckets